### PR TITLE
[Foundation] Fix superfluous scrollbars

### PR DESCRIPTION
**Pull request by floriank**
Meets https://community.openproject.org/work_packages/19249.

Also includes an update to the workflow form which previously did not use the new form styles.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://floriank-fix-19249-chrome-scrollbars.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://floriank-fix-19249-chrome-scrollbars.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/views/workflows/edit.html.erb [hunk 1]
```diff
@@ -29,30 +29,32 @@ See doc/COPYRIGHT.rdoc for more details.
...
+      <div class="form--column">
+        <div class="form--field -vertical">
+          <%= styled_label_tag :type_id, Type.model_name.human %>
+          <%= styled_select_tag 'type_id', options_from_collection_for_select(@types, "id", "name", @type && @type.id) %>
+        </div>
+      </div>
     </div>
-  </fieldset>
+    <%= submit_tag l(:button_edit), :name => nil, :accesskey => accesskey(:edit), class: 'button -highlight' %>
+  </div>
```
Comment (ulferts) [inline app/views/workflows/edit.html.erb]: This section used to be styled like a simple-filter just like (project, user, ...) so I don't think that it should be renamed.

**Comment by ulferts** [comment]
I am still convinced that it is wrong to no longer use the "simple-filter" styles here. They where created for exactly that purpose: https://community.openproject.org/assets/styleguide.html#simple-filters

**Comment by floriank** [comment]
I intentionally did not use them, as they do not provide an aligned layout - currently, the project administration uses these and it looks awful:

screenshot from 2015-03-24 15 48 14

**Comment by ulferts** [comment]
Then the fix IMO would be to correct the alignment problems for the simple--filter as this will reflect on all usages of it. There is just a higher leverage.

**Comment by floriank** [comment]
Then I shall go forth and changeth the css for that.

File: app/assets/stylesheets/_misc_legacy.sass [hunk 2]
```diff
@@ -274,7 +274,6 @@ ul.properties
     font-size: 120%
 
 .autoscroll
-  overflow-x: auto
```
Comment (ulferts) [inline app/assets/stylesheets/_misc_legacy.sass:277]: The `autoscroll` class is applied to other dom elements as well. As it is called `autoscroll` it should probably not loose the scrolling behaviour. I'd instead opt for removing the `autoscroll` class from the dom element.

File: app/views/workflows/edit.html.erb [hunk 3]
```diff
@@ -29,29 +29,31 @@ See doc/COPYRIGHT.rdoc for more details.
...
-        <%= select_tag 'type_id', options_from_collection_for_select(@types, "id", "name", @type && @type.id) %>
+      <li class="simple-filters-filter">
+        <%= styled_label_tag :role_id, Role.model_name.human, class: 'simple-filters--filter-name' %>
+        <div class="simple-filters--filter-value">
+          <%= styled_select_tag 'role_id', options_from_collection_for_select(@roles, "id", "name", @role && @role.id) %>
+        </div>
+        <div class="form--field">
+          <%= hidden_field_tag 'used_statuses_only', '0', id: 'used_statuses_only_hidden' %>
+
+          <%= styled_label_tag :used_statuses_only, class: 'form--label-with-check-box' do %>
```
Comment (ulferts) [inline app/views/workflows/edit.html.erb]: using the `styled_label_tag` helper here unfortunately adds the `form--label` class as well which does not play nicely with the `form--label-with-check-box` class as it limits the length of the label.

image

Options I see:

1) use `label_tag` instead
2) change the DOM slightly to use the `styled_label_tag` with the `-trailing-label` modifier class applied.

**Comment by NobodysNightmare** [comment]
Restarted some travis jobs, as github was not very responsive due to a [DDoS](https://status.github.com)

**Comment by floriank** [comment]
@NobodysNightmare thx!

**Pull request merged**